packages/flutter/lib/src/rendering/box.dart

```diff
@@ -524,27 +527,27 @@ class BoxConstraints extends Constraints {
         } else {
           whichFields = affectedFieldsList.single;
         }
-        throwError('BoxConstraints has ${affectedFieldsList.length == 1 ? 'a NaN value' : 'NaN values' } in $whichFields.');
+        throwError(ErrorSummary('BoxConstraints has ${affectedFieldsList.length == 1 ? 'a NaN value' : 'NaN values' } in $whichFields.'));
```

Comment:
We be we should spell out 'NaN' to make the message more beginner friendly.

Reply:
Added that to the list of items to cleanup with a followup CL.